static/app/components/growingInput.tsx

```diff
@@ -47,9 +47,26 @@ export const GrowingInput = forwardRef<HTMLInputElement, InputProps>(
 
     // If the input is controlled we resize it when the value prop changes
     useLayoutEffect(() => {
+      // This mutationObserver is used to resize the input if the inputRef's style changes
+      // It's necessary because the refs do not trigger useEffects
+      const observer = new MutationObserver(() => {
+        inputRef.current && resize(inputRef.current);
+      });
+
       if (isControlled && inputRef.current) {
         resize(inputRef.current);
       }
+
+      if (inputRef.current) {
+        observer.observe(inputRef.current, {
+          attributes: true,
+          attributeFilter: ['style'],
+        });
+      }
+
+      return () => {
+        observer.disconnect();
+      };
```

Comment:
This pretty much just monitors the ref for style changes. This is necessary because `resize()` takes into account the fontWeight, fontSize, and fontFamily styles to generate the correct width for the input. This previously caused issues because the tab title changes its fontWeight depending on if it's selected or not. This caused an annoying bug where tabs that were not initially selected, once selected, would see their tab title get cutoff slightly, since the tab title previously had lower weight, and the growingInput container did not resize the container accordingly. 

useEffect also does not trigger when refs change, so adding `inputRef.current?.style` to the dependency array did not achieve the desired behavior.

Reply:
It doesn't trigger when refs change, but it would if you did `props.style` or `props.className`. I actually think it might be simplest if you just modified the class and added `props.className` to the deps array.

Reply:
oml im an idiot for not thinking of this before 🤦‍♂️

Reply:
Going to add both classname and style to the dependency array since it looks like updating style rather than classname results in less jitter 

Updating classname:


https://github.com/user-attachments/assets/9152130d-bd26-403b-b056-fba8676e2944



Updating styles:

https://github.com/user-attachments/assets/7f9aa47c-2ae2-49e8-8287-c8d63cc12bf0

Reply:
Something to note about passing `style`, it looks like you are passing the object directly which will cause this resize to happen every time the component rerenders, event when `style` doesn't actually change. You'll need to wrap it in a `useMemo` to avoid that